### DISCUSSION TITLE
At what point would you separate / move routes in another package?

### DISCUSSION THREAD

**Discussion by fr3fou** [benbjohnson/wtf#30] [Q&A] [answered]
For simple projects having everything in the `http` package might work out, but at some point your handlers / routes count might be too much to store in one package. At what point would you consider doing something about that issue / what would you do?

**Comment by benbjohnson** [comment, marked as answer]
Good question. I don't find that I hit this limit in practice very often. I personally don't mind large files (1000-2000 [SLOC](https://en.wikipedia.org/wiki/Source_lines_of_code)) and I don't mind having a lot of files in a single package (~50) so I'd need to have 50KLOC in an `http` package before I'd consider splitting it out. I personally find it's more painful to split out into multiple packages than just having a single large package.

Another limiting factor is that there is a trend to break applications down to smaller separate applications as they grow large so I don't find that I have a single application that grows large enough to worry about it typically.

If you do have a mammoth application and need to split it out then I would probably just make subpackages under `http`. Maybe by some kind of subdomain? It's hard to say without a specific example and understanding what problem you're trying to solve by splitting.

**Reply by fr3fou** [reply]
let's say you wanted to split different packages for 
- user endpoints
- auth endpoints
- product endpoints
- shipping endpoints
- etc

**Reply by benbjohnson** [reply]
I don't personally find those to be good reasons for splitting up packages. You'll probably have some shared functionality and that'll end up having to move to an `http/httputil` package which then adds even more packages.
